### ISSUE TITLE
Regression in string concatenation

### ISSUE THREAD

**Issue by kiranchandramohan** [flang-compiler/flang#624]
There is a regression in string concatenation. This is regression is caused by the commit with title "Implement fixed- and deferred-length character in array constructor" and id 0918a2f2db72a90249e18ca062109c01b42d7ba9 causes regressions.

The following code on execution gives a segfault.
```
program char_test
  character(4), allocatable :: chars(:)
  chars = ['gary','stev']
  chars = 'flang'//chars
end program
```

I believe the if condition check in the following code (in the function get_ch_temp(DTYPE dtype) in the file tools/flang1/flang1exe/semutil2.c) is too strict and either has to be removed or amended.
```
        if (!ADD_DEFER(DTYPEG(sptr)) || ADJLENG(sptr))
          allocate_temp(sptr);
```

**Comment by gklimowicz** [comment]
@kiranchandramohan We think this was resolved with PR#639. Please let us know fi this is not the case with either this reproducer or another.

**Comment by kiranchandramohan** [comment]
Thanks @gklimowicz. I can confirm that this is fixed. Closing.
